### PR TITLE
Use new mutations createCurrencyApprovalTransaction and createCollectionApprovalTransaction

### PR DESCRIPTION
**Pull request by NicolasMahe**
### Project organization

- Related https://github.com/liteflow-labs/backend/issues/2402
- Dependant on https://github.com/liteflow-labs/backend/pull/2403

### Description

Use mutations `createCurrencyApprovalTransaction` and `createCollectionApprovalTransaction` in `useCreateOffer` and `useAcceptOffer`.

### Checklist

- [x] Update related changelogs

### CODE REVIEW

File: packages/hooks/src/useAcceptOffer.ts
Comment (NicolasMahe) [inline packages/hooks/src/useAcceptOffer.ts]:
```suggestion
    $quantity: Uint256!
```

File: packages/hooks/CHANGELOG.md
Comment (NicolasMahe) [inline packages/hooks/CHANGELOG.md]: @NicolasMahe backend is already released with this feature

File: packages/hooks/src/useCreateOffer.ts
Comment (antho1404) [inline packages/hooks/src/useCreateOffer.ts]: Now that these are dedicated mutations, instead of creating the queries in `useCreateOffer` and using it in both `useCreateOffer` and `useAcceptOffer`, let's instead have a dedicated (of even 2) hooks for approval that `useCreateOffer` and `useAcceptOffer` will use.

`useApproveCollection` and `useApproveCurrency`

Comment (NicolasMahe) [reply packages/hooks/src/useCreateOffer.ts]: Added but I need to properly test this.
The code is a lot more complex at the end...

Comment (NicolasMahe) [reply packages/hooks/src/useCreateOffer.ts]: Tested with the currency approval. The collection approval is a lot more tricky to test because the ERC contract we deploy using the rarible token deployer approve automatically the proxies.

File: packages/hooks/src/useApproveCollection.ts
Comment (NicolasMahe) [inline packages/hooks/src/useApproveCollection.ts:1]: Should I add this 2 hooks in the changelog even if they are not exported?

File: packages/hooks/src/useCreateOffer.ts
Comment (antho1404) [inline packages/hooks/src/useCreateOffer.ts:109]: Instead of dealing with 2 useEffect to try to synchronize a state, I think it would be better to have a `useMemo` as we already have all the information.

Maybe something like that (didn't test, just a quick write)

```ts
const [signing, setSigning] = useState(false)
  const activeStep = useMemo(() => {
    if (signing) return CreateOfferStep.SIGNATURE
    if (
      approveCurrencyActiveStep === ApproveCurrencyStep.PENDING ||
      approveCollectionActiveStep === ApproveCollectionStep.PENDING
    ) {
      return CreateOfferStep.APPROVAL_PENDING
    }
    if (
      approveCurrencyActiveStep === ApproveCurrencyStep.SIGNATURE ||
      approveCollectionActiveStep === ApproveCollectionStep.SIGNATURE
    ) {
      return CreateOfferStep.APPROVAL_SIGNATURE
    }
    return CreateOfferStep.INITIAL
  }, [signing, approveCurrencyActiveStep, approveCollectionActiveStep])
```

As "internally" we only need to know if it's initial state or signing state, this could be replaced by an simple boolean state

Similar comment for the rest.

Also, when using a `useEffect` we need to make sure to have the cleanup function when dealing with states

Comment (NicolasMahe) [reply packages/hooks/src/useCreateOffer.ts:109]: > Similar comment for the rest.

How does this apply to the `useAcceptOffer` hook? It's has more steps:
```
  TRANSACTION_SIGNATURE,
  TRANSACTION_PENDING,
  OWNERSHIP,
```
A new state with a boolean is not going to work.

Maybe the 2 useEffect could be merged into one with ifs rather than switch case.

> Also, when using a useEffect we need to make sure to have the cleanup function when dealing with states

This is on purpose. The useEffect should not change the step when it's cleanup when for example `approveCollectionActiveStep` change, otherwise it's going to reset the activeStep at the wrong time, which is not what we want if we are on the `SIGNATURE` step.

File: packages/hooks/src/useApproveCollection.ts
Comment (antho1404) [reply packages/hooks/src/useApproveCollection.ts:1]: let's not document then for now

File: packages/hooks/src/useCreateOffer.ts
Comment (antho1404) [reply packages/hooks/src/useCreateOffer.ts:109]: I created issue #130 about that, this is not blocking this PR, as everything works well. We could have local state and have a general memo "converting" all the states but let's do that in another step if needed as it is not required

**Review by antho1404 — APPROVED** [review]

**Pull request merged**
